sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -327,6 +329,14 @@ trait Row extends Serializable {
    */
   def getStruct(i: Int): Row = getAs[Row](i)
 
+  /**
+   * Returns value at position i of calendar Interval type as a
+   * [[org.apache.spark.sql.types.CalendarInterval]] object.
+   *
+   * @throws ClassCastException when data type does not match.
+   */
+  def getCalendarInterval(i: Int): CalendarInterval = CalendarInterval.fromString(get(i).toString)
```

Comment:
I am not sure I follow. Shouldn't this be a `CalendarInterval`? Or some java interval type? It seems a bit weird to make this public and put a string in a Row instead of the actual type.